Sources/MapboxNavigation/CarPlaySearchController+CPSearchTemplateDelegate.swift

```diff
@@ -6,69 +6,39 @@ import MapboxDirections
 
 @available(iOS 12.0, *)
 extension CarPlaySearchController: CPSearchTemplateDelegate {
-    static var recentItems = RecentItem.loadDefaults()
+    
+    // TODO: Find a way to get rid of these OR
+    // move button functions up and remove this extension
+    public func searchTemplate(_ searchTemplate: CPSearchTemplate, updatedSearchText searchText: String, completionHandler: @escaping ([CPListItem]) -> Void) {
+        delegate?.searchTemplate(searchTemplate, updatedSearchText: searchText, completionHandler: completionHandler)
+    }
+    
+    public func searchTemplate(_ searchTemplate: CPSearchTemplate, selectedResult item: CPListItem, completionHandler: @escaping () -> Void) {
+        delegate?.searchTemplate(searchTemplate, selectedResult: item, completionHandler: completionHandler)
+    }
+    
     public static let CarPlayGeocodedPlacemarkKey: String = "MBGecodedPlacemark"
     
     static var MaximumInitialSearchResults: UInt = 5
     static var MaximumExtendedSearchResults: UInt = 10
     /// A very coarse location manager used for focal location when searching
-    fileprivate static let coarseLocationManager: CLLocationManager = {
+    // TODO: Confirm that this location manager can become public to use in AppDelegate
+    public static let coarseLocationManager: CLLocationManager = {
```

Comment:
This CLLocationManager was fileprivate before. I made it public to access within AppDelegate+CarPlay.swift because it was a bad idea to initialize a new CLLocationManager in AppDelegate+CarPlay.swift. Should I actually do that instead of making this public?

Reply:
It’s only used for setting the `ForwardGeocodeOptions.focalLocation`, which is configured in the AppDelegate as of this PR. So it would be better to move this property to the AppDelegate. Once #2588 lands, it’ll become redundant to PassiveLocationManager-related notifications, so we’ll be able to remove it entirely.